Staging integration tests require redis just to import top-level conftest

## Problem
The staging integration-test job runs only deployed-service tests:
- `tests/integration/test_live_calculate.py`
- `tests/integration/test_live_economy.py`

But pytest still imports the top-level `tests/conftest.py`, and that file imports `redis` at module import time.

The workflow step currently installs only:
- `pytest`
- `httpx`

So test collection fails before any integration test runs with:

```text
ImportError while loading conftest '/home/runner/work/policyengine-api/policyengine-api/tests/conftest.py'.
tests/conftest.py:6: in <module>
    import redis
E   ModuleNotFoundError: No module named 'redis'
```

## Root cause
`redis` is only needed by the `rest_client` fixture, not by the deployed-service integration tests. But because it is imported eagerly at module import time, every test tier under `tests/` inherits that dependency.

## Fix
Import `redis` lazily inside the `rest_client` fixture so only tests that actually use that fixture require the package.

--- a/changelog.d/lazy-redis-import.fixed.md
+++ b/changelog.d/lazy-redis-import.fixed.md
@@ -1,0 +1,1 @@
+Fixed staging integration test collection by importing `redis` lazily inside the `rest_client` fixture instead of at `tests/conftest.py` module import time.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -3,7 +3,6 @@ import time
 from contextlib import contextmanager
 from subprocess import Popen, TimeoutExpired
 import sys
-import redis
 import pytest
 
 # Add the project root directory to PYTHONPATH
@@ -31,6 +30,7 @@ def running(process_arguments, seconds_to_wait_after_launch=0):
 def client():
     """run the app for the tests to run against"""
     from policyengine_api.api import app
+    import redis
 
     app.config["TESTING"] = True
     with running(["redis-server"], 3):

